### PR TITLE
feat(tree-item): allow disabling tree-item

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #3772 

## Summary

✨🌳🚫👈✨

### CODE REVIEW

**Comment by jcfranco** [comment]
@paulcpederson @macandcheese Could you give this a test drive and see if it's merge-worthy? 🧽

**Comment by macandcheese** [comment]
Noticed some weird focus behavior - seems possible to click to focus the first item only before the tree item sibling that is active is expanded. Once that is expanded, clicking the first disabled item no longer focuses it. It remains possible to keyboard focus the first item. The nested disabled item doesn't seem to be focusable at all.

Not sure which of the interactions makes the most sense from an a11y perspective but it seems a little inconsistent:

https://user-images.githubusercontent.com/4733155/186312416-14c3e095-30c8-494b-bd35-7dedbbab0ef3.mov

**Comment by jcfranco** [comment]
@macandcheese Fixed the issue you called out. Can you give it another try?

Focus styling was being applied regardless of component being disabled. We'll need to make sure any interactive component considers the `disabled` attribute for host styles using `:focus` since elements with `tabindex=-1` can still be focused via mouse click or `focus()`. cc @benelan @geospatialem

File: src/components/tree/tree.e2e.ts [hunk 1]
```diff
@@ -145,24 +143,58 @@ describe("calcite-tree", () => {
     expect(grandchildOne).not.toHaveAttribute("selected");
     expect(grandchildTwo).not.toHaveAttribute("selected");
 
-    await two.click();
+    // Puppeteer's element click will happen in the center of a component,
+    // so we call the method to ensure it happens on the component of interest
+
+    await two.callMethod("click");
```
Comment (eriklharper) [inline src/components/tree/tree.e2e.ts:149]: So cool, had no idea you could do this!

File: src/components/tree/tree.tsx [hunk 2]
```diff
@@ -246,94 +252,143 @@ export class Tree {
...
+      }
+
+      // When focus is on a root node that is also either an end node or a closed node, does nothing.
+      return;
+    }
+
+    if (event.key === "ArrowRight" && !target.disabled) {
+      if (target.hasChildren) {
+        if (target.expanded && getRootNode(this.el).activeElement === target) {
+          {
```
Comment (benelan) [inline src/components/tree/tree.tsx:305]: Just out of curiosity, what's the purpose of all these extra curly brackets

**Review by benelan — APPROVED** [review]
There may be extra curley brackets in the ` keyDownHandler` `event.key` conditionals, but if they have a purpose then the code LGTM

File: src/components/tree/tree.tsx (hunk 2 above)
Comment (jcfranco) [reply src/components/tree/tree.tsx:305]: To break the interwebs! 💻🔨

J/K, good catch. I'll remove these unnecessary extra blocks.

**Pull request merged**